### ISSUE TITLE
fail to bundle a single module

### ISSUE THREAD

**Issue by alexanderGalushka** [denoland/deno#4039]
@kitsonk @ry @bartlomieju submitting a separate issue brought up by @base698 in https://github.com/denoland/deno/pull/3828

worker.ts

```
function entryPoint(data) {
   console.log('received');
   postMessage("Hi 2 u");
}

if (self.name !== "workerFn") throw Error("Bad name");

onmessage = function(e) {
    entryPoint(e.data);
}

onerror = function(err) {
    console.error('on error', err);
    return false;
}
```

01_worker.ts

```
var [diagnostics, emit] = await Deno.bundle('./worker.ts');
const encoder = new TextEncoder();
Deno.writeFileSync('./worker.js', encoder.encode(emit));

const workerFn = new Worker('./worker.js', { name: 'workerFn', type: "module" });
workerFn.onmessage = console.log;

workerFn.postMessage('from client');
```

**Deno 0.32**

```
$ deno --allow-write 01_worker.ts
received
{ data: "Hi 2 u" }
```

**Deno 0.33**

```
$ deno --allow-write 01_worker.ts
Compile file:///home/justin/workspaces/deno-test/tests/01_worker.ts
error: Uncaught Error: Uncaught TypeError: Cannot destructure property 'deps' of 'mMap.get(...)' as it is undefined.
$deno$/workers.ts:169:17
    at poll ($deno$/workers.ts:169:17)
```

**Comment by kitsonk** [comment]
Ok, I think it is related to/duplicate of #4031, does adding `export {}` resolve the issue?

Also, is there a reason why you are bundling a single module?

**Comment by alexanderGalushka** [comment]
export{} doesn't solve an issue

**Comment by kitsonk** [comment]
Yeah, looking at #4031 it is a duplicate though.  So a) it does need `export {}` but then there is another bug after that which needs to be fixed.

**Comment by base698** [comment]
> Also, is there a reason why you are bundling a single module?

https://www.urbandictionary.com/define.php?term=You%27re%20Holding%20It%20Wrong

**Comment by kitsonk** [comment]
@base698 no, it is an honest query to understand a non-obvious use case.  When people use things in surprising ways it maybe because a) they don't understand or b) I don't understand.

Being glib doesn't really help anyone.

**Comment by base698** [comment]
We are trying to create separate workers.  So currently, unless something changed, you had to load them from disk so we are bundling to create a single file on disk that the worker can load each with it's own dependencies packaged into it separately.

**Comment by kitsonk** [comment]
> So currently, unless something changed, you had to load them from disk so we are bundling to create a single file on disk that the worker can load each with it's own dependencies packaged into it separately.

So it is currently a single module, but won't be eventually?

**Comment by alexanderGalushka** [comment]
it can either be a singe module or multiple modules, not deterministic

**Comment by kitsonk** [comment]
This was fixed in #4042
